ntbs-service-unit-tests/Helpers/EpisodesHelperTest.cs

```diff
@@ -58,17 +66,26 @@ public class EpisodeHelperTest
         };
 
         [Fact]
-        public void GroupTreatmentEventsByEpisode_GroupsBasedOnEndingOutcomeType()
+        public void GroupEpisodesIntoPeriods_GroupsBasedOnEndingOutcomeType()
         {
             // Act
-            var groupedEpisodes = _testTreatmentEvents.GroupByEpisode();
+            var periods = _testTreatmentEvents.GroupEpisodesIntoPeriods();
 
             // Assert
-            Assert.Equal(4, groupedEpisodes.Count);
-            Assert.Equal(2, groupedEpisodes[1].Count);
-            Assert.Single(groupedEpisodes[2]);
-            Assert.Equal(2, groupedEpisodes[3].Count);
-            Assert.Equal(3, groupedEpisodes[4].Count);
+            Assert.Equal(5, periods.Count);
+            Assert.Equal(new [] { _testTreatmentEvents[0], _testTreatmentEvents[1] }, periods[0].TreatmentEvents);
+            Assert.Equal(new [] { _testTreatmentEvents[2], }, periods[1].TreatmentEvents);
+            Assert.Equal(new [] { _testTreatmentEvents[3], }, periods[2].TreatmentEvents);
+            Assert.Equal(new [] { _testTreatmentEvents[5], _testTreatmentEvents[4] }, periods[3].TreatmentEvents);
+            Assert.Equal(new []
+            {
+                _testTreatmentEvents[6],
+                _testTreatmentEvents[8],
+                _testTreatmentEvents[7]
+            }, periods[4].TreatmentEvents);
+
+            Assert.Equal(new[] { false, true, false, false, false }, periods.Select(period => period.IsTransfer));
+            Assert.Equal(new[] { 1, 0, 2, 3, 4 }, periods.Select(period => period.PeriodNumber));
```

Comment:
I think it would be better to have a helper method like `AssertTreatmentPeriodMatchesExpected` or something like that, to keep the assertions for a single period together. Only a minor comment, but I think it would make it more obvious what is being tested.

Reply:
Yep I agree, I've changed that over!